src/librustdoc/core.rs

```diff
@@ -257,7 +257,7 @@ where
         .filter_map(|lint| {
             // Permit feature-gated lints to avoid feature errors when trying to
             // allow all lints.
-            if lint.name == warnings_lint_name || lint.feature_gate.is_some() {
+            if lint.name == warnings_lint_name || lint.feature_gate.is_some() || allowed_lints.iter().any(|l| lint.name == l) {
```

Comment:
I'm surprised: normally you don't need this check here... Maybe it's because it's part of `HardwiredLints`?

Reply:
Yeah, this is the thing I was confused about on discord. I'm not sure exactly why it works, just that it does.